src/utils/serve-functions.js

```diff
@@ -194,7 +204,19 @@ const createHandler = async function ({ dir, capabilities, omitFileChangesLog, w
     leading: false,
     trailing: true,
   })
-  watcher.on('change', debouncedOnChange).on('unlink', debouncedOnUnlink)
+  watcher
+    .on('change', debouncedOnChange)
+    .on('unlink', debouncedOnUnlink)
+    .on('add', async (path) => {
+      logBeforeAction({ omitLog: omitFileChangesLog, path, action: 'added' })
+
+      await watcher.unwatch(watchDirs)
+      ;({ functions, watchDirs } = await getFunctionsAndWatchDirs(dir))
```

Comment:
`functions` is used a few lines down when handling requests when searching for a matching function